common/plugins/add_endpoint.go

```diff
@@ -95,8 +97,15 @@ func SetupVethWithNames(lxcIfName, tmpIfName string, mtu int, ep *models.Endpoin
 	// Disable reverse path filter on the host side veth peer to allow
 	// container addresses to be used as source address when the linux
 	// stack performs routing.
-	args := []string{"-w", "net.ipv4.conf." + lxcIfName + ".rp_filter=0"}
-	_, err = exec.Command("sysctl", args...).CombinedOutput()
+	rpFilterPath := filepath.Join("/proc", "sys", "net", "ipv4", "conf", lxcIfName, "rp_filter")
+	f, err := os.OpenFile(rpFilterPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0666)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to open rp_filter configuration file of %s: %s",
+			lxcIfName, err)
+	}
+	std := exec.Command("echo", "0")
```

Comment:
Could we just open the file and write 4 zero-bytes directly into it? Then we don't need the additional exec with echo since this is doing the same, but writing 0 in there directly is more robust / less complicated. C equivalent I would just use `fd = open(path, O_WRONLY)` to open the procfs file.

Reply:
@borkmann the sysctl command uses the `os.O_WRONLY|os.O_CREATE|os.O_TRUNC`